src/gt4py/next/errors/exceptions.py

```diff
@@ -0,0 +1,111 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import textwrap
+from typing import Any, Optional
+
+from gt4py.eve import SourceLocation
+
+from . import formatting
+
+
+class CompilerError(Exception):
```

Comment:
I think this name is a bit ambiguous and it could be understood as an error in the compiler ("internal compiler error") instead of an error emitted by the compiler but caused by wrong input code from the user. I  propose to change it to something along the lines of `DSLError` or `GT4PySourceError`  or similar

Reply:
Agreed, that name is far from perfect, I'll put it up to a vote.